syntax = "proto3";
package Protocol;

import "Enum.proto";
import "Struct.proto";


message C_LOGIN
{

}

message S_LOGIN
{
	bool success = 1;
	repeated ObjectInfo players = 2;	// 아이디 발급 전
}

message C_ENTER_GAME
{
	uint64 playerIndex = 1;
}

message S_ENTER_GAME
{
	bool success = 1;
	ObjectInfo player = 2;
}

message C_LEAVE_GAME
{

}

message S_LEAVE_GAME
{

}

message S_SPAWN
{
	repeated ObjectInfo objects = 1;
}

message S_DESPAWN
{
	repeated uint64 object_ids = 1;
}

message C_TURN
{
	uint64 object_id = 1;
	float yaw = 2;
}

message S_TURN
{
	uint64 object_id = 1;
	float yaw = 2;
}

message C_JUMP
{
	uint64 object_id = 1;
}

message S_JUMP
{
	uint64 object_id = 1;
}

message C_MOVE
{
	PosInfo info = 1;
}

message S_MOVE
{
	PosInfo info = 1;
}

message C_CHAT
{
	string msg = 1;
}

message S_CHAT
{
	uint64 playerId = 1;
	string msg = 2;
}

message C_FIRE
{
	
}

message S_FIRE
{
	uint64 object_id = 1;
}

message C_SNIPER_FIRE
{
	uint64 object_id = 1;
	VectorInfo start = 2;
	VectorInfo end = 3;
}

message S_SNIPER_FIRE
{
	uint64 object_id = 1;
	VectorInfo start = 2;
	VectorInfo end = 3;
}

message C_CHANGE_ITEM
{
	uint64 object_id = 1;
	uint32 key = 2;
}

message S_CHANGE_ITEM
{
	uint64 object_id = 1;
	uint32 key = 2;
}